download.sh

```diff
@@ -4,12 +4,20 @@ set -e
 if [[ "$OSTYPE" == "linux-gnu" ]]; then
     echo 'downloading linux c-server-sdk shared library'
     rm -rf 'linux-gcc-64bit-dynamic.tar' 'include' 'lib'
-    curl -s -o 'linux-gcc-64bit-dynamic.tar' -L 'https://github.com/launchdarkly/c-server-sdk/releases/download/2.2.1/linux-gcc-64bit-dynamic.tar'
+    curl https://api.github.com/repos/launchdarkly/c-server-sdk/releases/latest | \
+    grep browser_download_url | \
+    grep linux-gcc-64bit-dynamic.tar | \
+    cut -d '"' -f 4 | \
+    xargs curl -s -o 'linux-gcc-64bit-dynamic.tar' -L
     tar -xvf 'linux-gcc-64bit-dynamic.tar'
 elif [[ "$OSTYPE" == "darwin"* ]]; then
     echo 'downloading osx c-server-sdk shared library'
     rm -rf 'osx-clang-64bit-dynamic.tar' 'include' 'lib'
-    curl -s -o "osx-clang-64bit-dynamic.tar" -L "https://github.com/launchdarkly/c-server-sdk/releases/download/2.2.1/osx-clang-64bit-dynamic.tar"
+    curl https://api.github.com/repos/launchdarkly/c-server-sdk/releases/latest | \
+    grep browser_download_url | \
+    grep osx-clang-64bit-dynamic.tar | \
+    cut -d '"' -f 4 | \
+    xargs curl -s -o 'osx-clang-64bit-dynamic.tar' -L
     tar -xvf 'osx-clang-64bit-dynamic.tar'
```

Comment:
You could refactor this into a bash function which takes a single filename parameter (`linux-gcc-64bit-dynamic.tar` | `osx-clang-64bit-dynamic.tar`). Up to you if you want to do so.